src/devices/Tsl256x/README.md

```diff
@@ -0,0 +1,110 @@
+
+The TSL2560 device permits an SMB-Alert style interrupt, and the TSL2561 device supports a traditional level style interrupt that remains asserted until the firmware
+clears it.
+
+## Basic usage
+
+TSL2560 and TSL2561 are designed to have an integration time and a gain for measuring the 2 ADC. The defaults are 402 milliseconds and a normal gain of x1.
+
+The basic usage is the following:
+
+```csharp
+I2cDevice i2cDevice = I2cDevice.Create(new I2cConnectionSettings(1, Tsl256x.DefaultI2cAddress));
+Tsl256x tsl256X = new(i2cDevice, PackageType.Other);
+tsl256X.IntegrationTime = IntegrationTime.Integration402Milliseconds;
+tsl256X.Gain = Gain.Normal;
+var lux = tsl256X.MeasureAndGetIlluminance();
+Console.WriteLine($"Illuminance is {lux.Lux} Lux");
+```
+
+**notes**:
+
+- Be aware, there are 2 types of packaging the CS and the Others T, FN and CL. Refer to the documentation to understand which package you hav on the board. This is an argument when creating the sensor as the calculation for the illuminance is different.
+- There are 3 different possible I2C addresses for this device depending on how the address pin is setup:
+  - DefaultI2cAddress = 0x29: When address pin is to the ground
+  - SecondI2cAddress = 0x39: When address pin is floating
+  - ThirdI2cAddress = 0x49: When address pin is to VDD
+
+## Check the version
+
+You can determine if you have a TSL2560 or TSL2561 version:
+
+```csharp
+var ver = tsl256X.Version;
+string msg = ver.Major & 0x01 == 0x01 ? $"This is a TSL2561, version {ver}" : $"This is a TSL2560, version {ver}";
+Console.WriteLine(msg);
+```
+
+## Using interruptions
+
+You can set interruptions, you have different possible ones you can set thru the `InterruptLevel`:
+
+- OutputDisabled: will disable the interrupt
+- LevelInterrupt: will use the permanent interrupt you'll setup, see right after
+- SmbAlertCompliant: will send an alert on the SMB bus (version TSL2560 only) on address 0b0000_1100
+- TestMode: will generate an interruption for test purpose
+
+The `InterruptPersistence` will allow you to select a specific timing for the interrupt:
+
+- EveryAdc: each time you have a measurement
+- AnyValueOutsideThreshold: any time a value will be outside of the setup threshold
+- OutOfRangeXXIntegrationTimePeriods: where XX is from 2 to 15, will interrupt only if the value remains outside more than XX times
+
+The following example shows how to setup a basic interruption using test mode:
+
+```csharp
+Console.WriteLine("Set interruption to test. Read the interrupt pin");
+GpioController controller = new();
+controller.OpenPin(PinInterrupt, PinMode.Input);
+tsl256X.InterruptControl = InterruptControl.TestMode;
+tsl256X.Power = true;
```

Comment:
I'm confused what Power does? Could you rename it to something less confusing? (i.e. Disabled if it means to turn on/off device)

Reply:
Yes, will rename. Wanted to keep close to datasheet but Enabled will be much better